cclib/bridge/cclib2pyscf.py

```diff
@@ -112,4 +129,491 @@ def makepyscf_mos(ccdata, mol):
+    )
+    attributes["atomcharges"] = {"mulliken": mulliken_charges, "lowdin": lowdin_charges}
+
+    # Excited states.
+    if len(et) > 0:
+        # PySCF tracks convergence for each state which is great.
+        attributes["metadata"]["success"] = all(
+            itertools.chain(*(etmethod.converged for etmethod in et))
+        )
+
+        # In cclib 1.x, 'energies' are actually expected to be cm-1. In 2.x, this will change to Hartree.
+        attributes["etenergies"] = list(itertools.chain(*(etmethod.e for etmethod in et)))
+        attributes["etoscs"] = list(
+            itertools.chain(*(etmethod.oscillator_strength(gauge="length") for etmethod in et))
+        )  # or do we want velocity?
+        # et.analyse() prints real symmetries, so they must be available somewhere...
+
+        attributes["etsyms"] = []
+        for etmethod in et:
+            if isinstance(etmethod, pyscf.tdscf.rks.TDA):
+                attributes["metadata"]["excited_states_method"] = "TDA"
+
+            elif isinstance(etmethod, pyscf.tdscf.rks.TDDFT):
+                attributes["metadata"]["excited_states_method"] = "TD-DFT"
+
+            elif isinstance(etmethod, pyscf.tdscf.rhf.TDA):
+                attributes["metadata"]["excited_states_method"] = "CIS"
+
+            elif isinstance(etmethod, pyscf.tdscf.rhf.TDHF):
+                attributes["metadata"]["excited_states_method"] = "RPA"
+
+            else:
+                attributes["metadata"]["excited_states_method"] = type(etmethod).__name__
+
+            # From from pyscf/pyscf/tdscf/rhf.py
+            # import pyscf.symm
+            # orbsym = scf.get_orbsym()
+            # x_sym = pyscf.symm.direct_prod(orbsym[mo_occ==2], orbsym[mo_occ==0], mol.groupname)
+
+            # Need to be careful in interpreting the multiplicity of the excited state.
+            if etmethod.singlet and mol.multiplicity != 2:
+                et_mult = "Singlet"
+
+            elif etmethod.singlet:
+                et_mult = "Doublet"
+
+            else:
+                et_mult = "Triplet"
+
+            attributes["etsyms"].extend([et_mult for i in range(len(etmethod.e))])
+
+        # Orbital contributions.
+        # In PySCF, occupied and virtual orbital indices are stored separately.
+        attributes["etsecs"] = []
+
+        for index in range(len(attributes["etenergies"])):
+            attributes["etsecs"].append([])
+
+            # Assuming x are excitations, y are de-excitations.
+            # y is ignored for now.
+            x, y = list(itertools.chain(*(etmethod.xy for etmethod in et)))[index]
+
+            if not scf.istype("UHF"):
+                # The coefficients of x (and presumably also y) are normalised to 0.5,
+                # but we expect 1.0
+                # Renormalize.
+                # Taken from pyscf.tdscf.rhf.get_nto()
+                #
+                # Would appreciate someone checking this makes sense?
+                x *= 1.0 / np.linalg.norm(x)
```

Comment:
IIRC I ripped this from pyscf many years ago, but it indicates that for CIS/TDA you're correct, but if there are deexcitation vectors you need to normalize them together:
```python
    def norm_xy(z: np.ndarray, nocc: int, nvirt: int) -> Tuple[np.ndarray, np.ndarray]:
        x, y = z.reshape(2, nvirt, nocc)
        norm = 2 * (np.linalg.norm(x) ** 2 - np.linalg.norm(y) ** 2)
        norm = 1 / np.sqrt(norm)
        return (x * norm).flatten(), (y * norm).flatten()
```